src/data_types/indexed_tensor.hpp

```diff
@@ -152,6 +152,39 @@ KOKKOS_FUNCTION void internal_tensor_mul(
     ((internal_tensor_mul_elem<GlobalTensorIndexIdMap, Is>(result, t...)), ...);
 }
 
+template <char ID, class AllIndexIdMaps>
+KOKKOS_INLINE_FUNCTION void check_id_validity()
+{
+    constexpr std::size_t n_occurrences
+            = char_occurrences_v<ID, index_identifiers_t<AllIndexIdMaps>>;
+    if constexpr (n_occurrences == 2) {
+        using RelevantVectorIndexSets = relevant_vector_index_sets_t<ID, AllIndexIdMaps>;
+        using Set1 = ddc::type_seq_element_t<0, RelevantVectorIndexSets>;
+        using Set2 = ddc::type_seq_element_t<1, RelevantVectorIndexSets>;
+        constexpr bool has_covariant_idx = (is_covariant_vector_index_set_v<Set1>)
+                                           or (is_covariant_vector_index_set_v<Set2>);
+        constexpr bool has_contravariant_idx = (is_contravariant_vector_index_set_v<Set1>)
+                                               or (is_contravariant_vector_index_set_v<Set2>);
+        static_assert(
+                has_covariant_idx and has_contravariant_idx,
+                "Repeated indices should not be associated with two covariant or two contravariant "
+                "indices.");
```

Comment:
I think we should avoid mixing notations for operators like `&&`/`and`, `||`/`or` etc

Reply:
Fixed  ce2738e